[Clang] Fix LibTooling doc

maxmosk opened the pull request:
Replace CommonOptionsParser ctor by factory method ::create.

Found on page https://clang.llvm.org/docs/LibTooling.html

Sirraide commented on clang/docs/LibTooling.rst:
```diff
@@ -71,9 +71,9 @@ and automatic location of the compilation database using source files paths.
     // CommonOptionsParser constructor will parse arguments and create a
     // CompilationDatabase.  In case of error it will terminate the program.
-    CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
+    auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
Also, `->` needs to be replaced with `.` again below.

Sirraide commented on clang/docs/LibTooling.rst:
```diff
@@ -133,9 +133,9 @@ version of this example tool is also checked into the clang tree at
-    CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
-    ClangTool Tool(OptionsParser.getCompilations(),
-                   OptionsParser.getSourcePathList());
+    auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
As above

Sirraide commented on clang/include/clang/Tooling/CommonOptionsParser.h:
```diff
@@ -56,9 +56,9 @@ namespace tooling {
-///   CommonOptionsParser OptionsParser(argc, argv, MyToolCategory);
-///   ClangTool Tool(OptionsParser.getCompilations(),
-///                  OptionsParser.getSourcePathList());
+///   auto OptionsParser = CommonOptionsParser::create(argc, argv, MyToolCategory);
```
And here too

Sirraide reviewed (changes requested):
Thanks for improving our documentation!

There’s some error handling missing here. The `CommonOptionsParser` constructor would previously just exit, but that is no longer the case here, so we actually need to handle errors now.

Sirraide commented:
I also just noticed that there already is a pr for this: #70427.

maxmosk commented:
> I also just noticed that there already is a pr for this: #70427.

This PR wasn't approved and doc is still not fixed :(
What to do with this PR?

Sirraide commented:
> This PR wasn't approved and doc is still not fixed :( What to do with this PR?

It somehow just went unnoticed all this time, unfortunately. I’ll approve the other pr since it seems to do the same thing that we’re also doing somewhere else in our documentation, and it’s pretty much what I suggested here anyway.

The comment in `clang/Tooling/CommonOptionsParser.h` still needs to be updated, which the other pr doesn’t do; you can still do that if you’d like. Sorry about the entire confusion here; this is something that shouldn’t normally happen...

maxmosk committed:
[Clang][Docs] Synchronize the LibTooling example

Synchronize the example in LibTooling documentation and header CommonOptionsParser.h

maxmosk commented:
@Sirraide updated with header doc fix, please review

Sirraide reviewed (approved):
LGTM seeing as this matches what’s already in the documentation.

Sirraide commented:
Right, what `clang-format` is suggesting here is a bit bizzare; please reformat the comment manually so every line fits within the 80 column limit.
